tests/unittests/test_dmi.py

```diff
@@ -80,29 +68,29 @@ def _sysctl_subp(cmd) -> SubpResult:
                 raise subp.ProcessExecutionError()
             return SubpResult(content, error)
 
-        self.patched_funcs.enter_context(
-            mock.patch("cloudinit.dmi.subp.subp", side_effect=_sysctl_subp)
-        )
+        mocker.patch("cloudinit.dmi.subp.subp", side_effect=_sysctl_subp)
 
-    def patch_mapping(self, new_mapping):
-        self.patched_funcs.enter_context(
-            mock.patch("cloudinit.dmi.DMIDECODE_TO_KERNEL", new_mapping)
-        )
+    def patch_mapping(self, mocker, new_mapping):
+        mocker.patch("cloudinit.dmi.DMIDECODE_TO_KERNEL", new_mapping)
```

Comment:
Do we really need this one-liner helper method? Can we just drop it and inline the mocker.patch call at the 4 call-sites? Only one of those sites provides a non-empty dict.